flask_resources/errors.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# Flask-Resources is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Exceptions used in Flask Resources module."""
+
+import json
+
+from flask import g
+from werkzeug.exceptions import HTTPException
+from werkzeug.http import http_date
+
+# FIXME: (LARS) This has to be reviewed
+
+
+class RESTException(HTTPException):
+    """HTTP Exception delivering JSON error responses."""
+
+    errors = None
+
+    def __init__(self, errors=None, **kwargs):
+        """Initialize RESTException."""
+        super(RESTException, self).__init__(**kwargs)
+        if errors is not None:
+            self.errors = errors
+
+    def get_errors(self):
+        """Get errors.
+
+        :returns: A list containing a dictionary representing the errors.
+        """
+        return [e.to_dict() for e in self.errors] if self.errors else None
+
+    def get_description(self, environ=None):
+        """Get the description."""
+        return self.description
+
+    def get_body(self, environ=None):
+        """Get the request body."""
+        body = dict(status=self.code, message=self.get_description(environ),)
+
+        errors = self.get_errors()
+        if self.errors:
+            body["errors"] = errors
+
+        if self.code and (self.code >= 500) and hasattr(g, "sentry_event_id"):
+            body["error_id"] = str(g.sentry_event_id)
+
+        return json.dumps(body)
+
+    def get_headers(self, environ=None):
+        """Get a list of headers."""
+        return [("Content-Type", "application/json")]
+
+
+class FieldError(object):
+    """Represents a field level error.
+
+    .. note:: This is not an actual exception.
+    """
+
+    def __init__(self, field, message, code=None):
+        """Init object.
+
+        :param field: Field name.
+        :param message: The text message to show.
+        :param code: The HTTP status to return. (Default: ``None``)
+        """
+        self.res = dict(field=field, message=message)
+        if code:
+            self.res["code"] = code
+
+    def to_dict(self):
+        """Convert to dictionary.
+
+        :returns: A dictionary with field, message and, if initialized, the
+            HTTP status code.
+        """
+        return self.res
+
+
+#
+# Search
+#
+class SearchPaginationRESTError(RESTException):
+    """Search pagination error."""
+
+    code = 400
+
+    def __init__(self, errors=None, **kwargs):
+        """Initialize exception."""
+        _errors = []
+        if errors:
+            for field, messages in errors.items():
+                _errors.extend([FieldError(field, msg) for msg in messages])
+        super(SearchPaginationRESTError, self).__init__(
+            errors=_errors, **kwargs
+        )
+
+
```

Comment:
Responsibility wise: this deals with abstracting a representation of the errors in the Flask views, so that they can be serialized in a second phase... right? But there seems to be coupling to json serialization right now. The JSON serialization will be separate later right?

Reply:
Right now errors are there because they are needed for this to work, but they have not been worked on, mainly migrated. See comment at the bottom of the file:

> # FIXME: (LARS) This has to be reviewed

This means: You are totally right, but they will change a lot. Not just now since they have not been worked on 👼

Reply:
https://github.com/inveniosoftware/flask-resources/issues/3